### PR TITLE
Add function to report Agent time usage to Stripe

### PR DESCRIPTION
**Pull request by satococoa**
## Summary

This PR introduces a new mechanism to report usage-based billing metrics to Stripe.

**Key changes:**

1. **Database Schema Updates**  
   - Changed the column type from `integer` to `numeric` in `agentActivities` to handle larger values that may exceed the 32-bit integer limit.
     - And the column's name in TypeScript was wrong, I've fixed.
   - Introduced a new `agent_time_usage_reports` table to track accumulated agent usage time on a per-team, per-billing-cycle basis.

2. **Usage-based Billing Logic**  
   - Added a new `AgentTimeUsageDAO` class that encapsulates database operations related to agent time usage, including transactional updates.
   - Implemented the `processUnreportedActivities` function that identifies unprocessed agent activities within a given billing cycle, calculates the incremental usage in minutes, and reports those increments to Stripe via their `v2.billing.meterEvents` API.

3. **Integration with Stripe**  
   - Upgraded the Stripe API version and library version to ensure compatibility with the new events API.
     - `stripe.v2.billing.meterEvents.` was not in previous version.
   - When new usage increments are detected, we create a meter event in Stripe to reflect the additional chargeable minutes.

4. **Unit Tests**  
   - Added unit tests (`agent-time-usage.test.ts`) to cover edge cases, including boundaries at exactly 60 seconds and slightly under a full minute.
   - Tested scenarios such as no unreported activities, increments less than a minute (which results in no billing update), and handling data across multiple billing periods.

---

### The `processUnreportedActivities` Function Explained

`processUnreportedActivities` is the core of this PR and is responsible for tying together the data retrieval, calculation, and reporting logic.

**What does it do?**

- Determines the correct billing cycle based on the current subscription data and a given `targetDate`.
- Fetches all agent activities that have not yet been associated with a usage report within that billing cycle.
- Calculates the total accumulated duration of all these activities and determines how many **additional whole minutes** have accrued since the last report.  
  *(We track usage in whole minutes because our Stripe product is configured to bill on a per-minute basis.)*
- If there are new billable increments, creates a usage report, marks activities as processed, and reports the increment to Stripe as a meter event.

**How does it work?**

```mermaid
flowchart TD
    A[Determine Billing Cycle] --> B[Fetch Unreported Activities]
    B --> |Activities Found| C[Calculate Total Usage Duration]
    B --> |No Activities| J[Return no changes]
    C --> D[Get Last Usage Report]
    D --> E[Calculate Increment in Whole Minutes]
    E --> |Increment > 0| F[Create Stripe Meter Event]
    E --> |Increment = 0| J[Return no changes]
    F --> G[Create Usage Report in DB]
    G --> H[Mark Activities as Processed]
    H --> I[Return Report ID]
```

**Key Points in the Flow:**
- Accumulate Duration: Sums totalDurationMs from all unreported activities.
- Determine Increment: Converts the accumulated duration into whole minutes and compares it to the previously reported total minutes.
- Meter Event Creation: If there’s at least a 1-minute increment, we send that increment to Stripe.
- Database Consistency: Creates a usage report and updates activities to ensure no double counting in subsequent runs.

----

## Additional Notes
- We currently retrieve the customerId from Stripe subscriptions each time. In the future, we may store it in our DB to reduce external API calls.
- Using numeric for totalDurationMs avoids overflow issues with extremely large durations.
- The included tests cover various edge cases, ensuring that increments are calculated accurately.

## What I didn't do
- This function is not called anywhere.
- I will implement it in the following PRs.

## TODO
- [x] Ensure the Stripe update does not affect other areas.
- [x] Run migration for Preview
- [ ] Run migration for Production

### CODE REVIEW

**Review by toyamarinyon — APPROVED** [review]

File: services/external/stripe/config.ts [hunk 1]
```diff
@@ -2,5 +2,5 @@ import { Stripe } from "stripe";
 
 export const stripe = new Stripe(process.env.STRIPE_SECRET_KEY as string, {
 	// https://github.com/stripe/stripe-node#configuration
-	apiVersion: "2024-06-20",
+	apiVersion: "2024-11-20.acacia",
```
Comment (shige) [inline services/external/stripe/config.ts:5]: 📝 For My Notes
- 🔗 https://docs.stripe.com/changelog#2024-11-20.acacia

**Comment by satococoa** [comment]
Thank you for your reviews.

@shige Could you run migration and deploy for Production?

**Comment by shige** [comment]
> Could you run migration and deploy for Production?

@satococoa 
I did it! `bun drizzle-kit migrate` 🚀
